hirundo/dataset_optimization.py

```diff
@@ -97,78 +106,227 @@ class DatasetOptimizationResults(BaseModel):
 }
 
 
+class Metadata(BaseModel, ABC):
```

Comment:
Please take a look at my comment regarding `classes` being a field of `Metadata` in the `core` PR.